From: 崔家金 <cuijiajin186@126.com>
#Subject: [PATCH] user32: Make text warp on display.
Message-Id: <fc7af4b.172e.176db4a4887.Coremail.cuijiajin186@126.com>
Date: Thu, 7 Jan 2021 13:20:47 +0800 (GMT+08:00)

From d9b29cc32779006d91792f645d6d54f9aee9e5ea Mon Sep 17 00:00:00 2001
From: Jiajin Cui <cuijiajin@uniontech.com>
Date: Thu, 7 Jan 2021 10:32:30 +0800
Subject: [PATCH] user32: Make text warp on display.

In the official document of microsoft, uiLengthDrawn param Receives the number of characters processed by DrawTextEx, including white-space characters.

Signed-off-by: Jiajin Cui <cuijiajin@uniontech.com>
---
 dlls/user32/text.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/dlls/user32/text.c b/dlls/user32/text.c
index 11111111111..11111111111 100644
--- a/dlls/user32/text.c
+++ b/dlls/user32/text.c
@@ -988,6 +988,9 @@ INT WINAPI DrawTextExW( HDC hdc, LPWSTR str, INT i_count,
                 y = rect->bottom + (invert_y ? 0 : -size.cy);
         }
 
+    if (dtp)
+        dtp->uiLengthDrawn += len;
+
 	if (!(flags & DT_CALCRECT))
 	{
             const WCHAR *str = line;
@@ -1044,8 +1047,6 @@ INT WINAPI DrawTextExW( HDC hdc, LPWSTR str, INT i_count,
 	    max_width = size.cx;
 
         y += invert_y ? -lh : lh;
-        if (dtp)
-            dtp->uiLengthDrawn += len;
     }
     while (strPtr && !last_line);
 

-- 
2.39.0

